packages/components/src/internal/components/labels/LabelsConfigurationPanel.tsx

```diff
@@ -0,0 +1,352 @@
+                            >
+                                <span className="fa fa-trash" />
+                                <span>&nbsp;Delete</span>
+                            </DisableableButton>
+                        )}
+                        {isNew && (
+                            <Button bsStyle="default" disabled={saving} onClick={onCancel}>
+                                Cancel
+                            </Button>
+                        )}
+                        <Button bsStyle="success" className="pull-right" disabled={!dirty || saving} onClick={onSave}>
+                            {saving ? 'Saving...' : 'Save'}
+                        </Button>
+                    </div>
+                </form>
+            )}
+            {showDeleteConfirm && (
+                <ConfirmModal
+                    cancelButtonText="Cancel"
+                    confirmButtonText="Yes, Delete"
+                    title="Delete Label"
+                    onCancel={onToggleDeleteConfirm}
+                    onConfirm={onConfirmDelete}
+                >
+                    <div>
+                        The <b>{updatedTemplate.name}</b> label will be deleted.
+                        <strong>This cannot be undone.</strong> Do you wish to proceed?
+                    </div>
+                </ConfirmModal>
+            )}
+        </>
+    );
+});
+
+export const LabelsConfigurationPanel: FC<LabelTemplatesPanelProps> = memo(props => {
+    const { api, setIsDirty } = props;
+    const [templates, setTemplates] = useState<LabelTemplate[]>([]);
+    const [error, setError] = useState<string>();
+    const [selected, setSelected] = useState<number>();
+    const addNew = useMemo(() => selected === NEW_LABEL_INDEX, [selected]);
+
+    const queryLabelTemplates = useCallback(
+        (newLabelTemplate?: number) => {
+            setError(undefined);
+
+            api.labelprinting
+                .getLabelTemplates()
+                .then(labelTemplates => {
+                    setTemplates(labelTemplates);
+
+                    if (newLabelTemplate)
+                        setSelected(labelTemplates.findIndex(template => template.rowId === newLabelTemplate));
+                })
+                .catch(reason => {
+                    if (reason.status === 404) {
+                        // try to create list
+                        api.labelprinting
+                            .ensureLabelTemplatesList()
+                            .then(() => queryLabelTemplates(newLabelTemplate))
+                            .catch(newReason => {
+                                console.error(newReason);
+                                setError('Error: Failed to create template list.');
+                            });
+                    } else {
+                        console.error(reason);
+                        setError('Error: Unable to load label templates.');
+                    }
+                });
```

Comment:
why not just call the `ensureLabelTemplatesList` first here? it does this logic for us (i.e. it tries to call getLabelTemplates and then falls back to calling `createLabelTemplateList`). Seems like this sequence is duplicating that effort.

Reply:
agreed, replaced with ensure method